### PR TITLE
Migrate to pnpm for package management

### PR DESCRIPTION
**Pull request by TrySound**
- setup corepack for pnpm and forbid yarn usage with "engines" field
- converted yarn lockfile into pnpm lockfile with `pnpm import` command
- replaced yarn with pnpm in scripts and github actions, had to use custom cache to support corepack

This PR is only migration. Will use new pnpm features in other PRs.

```diff
du -ckd1 node_modules (yarn)
- 1125032

du -ckd1 node_modules/.pnpm
+ 857456
```

Install time (including cache downloads) in github actions reduced from 42s to 23s

## Before requesting a review

- [ ] if the PR is WIP - use draft mode
- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")
- [ ] what kind of review is needed?
  - [x] conceptual
  - [X] detailed
  - [ ] with testing

## Test cases

- [ ] step by step interaction description and what is expected to happen

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio-designer/blob/main/apps/designer/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `designer/env-check.js` if mandatory

### CODE REVIEW

**Review by istarkov — APPROVED** [review]

File: .github/actions/ci-setup/action.yml [hunk 1]
```diff
@@ -8,8 +8,22 @@ runs:
       uses: actions/setup-node@v3
       with:
         node-version: 16
-        cache: "yarn"
+
+    - run: corepack enable
```
Comment (istarkov) [inline .github/actions/ci-setup/action.yml:12]: Do we need to run this locally?

File: apps/designer/package.json [hunk 2]
```diff
@@ -9,7 +9,7 @@
     "typecheck": "tsc",
     "lint": "eslint . --ext .ts,.tsx --max-warnings 0",
     "test": "NODE_OPTIONS=--experimental-vm-modules jest",
-    "checks": "yarn lint && yarn typecheck && yarn prettier --check && yarn test",
+    "checks": "pnpm lint && pnpm typecheck && pnpm prettier --check && pnpm test",
```
Comment (istarkov) [inline apps/designer/package.json:12]: all can be run in parallel as looks independent

File: .github/actions/ci-setup/action.yml (hunk 1 above)
Comment (TrySound) [reply .github/actions/ci-setup/action.yml:12]: If you don't have pnpm istalled locally

File: apps/designer/package.json (hunk 2 above)
Comment (TrySound) [reply apps/designer/package.json:12]: Yes, will propose some solution

**Review by kof — APPROVED** [review]
🎉

**Comment by andrasbacsai** [comment]
LGTM! Will it break all existing preview deployments in Vercel as we need to change the build command from `yarn` to `pnpm`?

**Review by andrasbacsai — APPROVED** [review]
LGTM! 
Will it break old branches preview deployments in Vercel as we need to change the build command from `yarn` to `pnpm`?

**Comment by TrySound** [comment]
Yes, it will. Though we not merging yet this because of storybook setup which need to be additionally configured.

**Comment by kof** [comment]
@TrySound do you wanna let @Andarist help us with storybook and pnpm here?

**Comment by kof** [comment]
This PR hanging is no good, we need to merge it

File: package.json [hunk 3]
```diff
@@ -31,20 +24,26 @@
...
     "prettier": "2.7.1",
+    "tsx": "^3.9.0",
     "turbo": "1.3.1",
     "typescript": "4.7.4"
   },
   "engines": {
     "node": "16",
-    "yarn": "1.22"
+    "pnpm": "7.18",
+    "yarn": "This project is configured to use pnpm"
```
Comment (Andarist) [inline package.json:35]: out of curiosity - what this is for? 🤔

**Review by Andarist — APPROVED** [review]

File: package.json (hunk 3 above)
Comment (TrySound) [reply package.json:35]: It will show the message when user try to run yarn without corepack enabled

File: .npmrc [hunk 4]
```diff
@@ -0,0 +1 @@
+node-linker=hoisted
```
Comment (Andarist) [inline .npmrc:1]: image

🤣 

This kinda sums up most of my experience with non-classic `node_modules` 😅 

out of curiosity - what kind of an effect does it have on install times with pnpm?

File: apps/designer/package.json [hunk 5]
```diff
@@ -22,6 +22,7 @@
     "@lexical/react": "^0.6.0",
     "@lexical/selection": "^0.6.0",
     "@lexical/utils": "^0.6.0",
+    "@prisma/client": "4.3.1",
```
Comment (Andarist) [inline apps/designer/package.json:25]: q: why is this needed in the `designer`? Shouldn't it reach to the `@prisma/client` only through `@webstudio-is/prisma-client`?

Comment (Andarist) [reply apps/designer/package.json:25]: For some other reasons it's likely that you have to use this in the `@webstudio-is/prisma-client`:
```prisma
generator client {
  output          = "../src/generated-client"
}
```

and perhaps that would fix your problems here. I have a draft for this change locally but didn't yet have time to finish it up.

Comment (TrySound) [reply apps/designer/package.json:25]: See, remix bundles all its packages because of esm. Though with pnpm all packages dependencies are no longer available in designer. So prisma/client was a check. So I decided to use hoisted until remix issues are solved.

Comment (TrySound) [reply apps/designer/package.json:25]: If we use custom output remix will try to bundle generated client
